students/DrewSmith/lessons/lesson10/assignment/databse.py

```diff
@@ -0,0 +1,204 @@
+'''
+HP Norton inventory and customer information database functions
+'''
+
+
+import csv
+import logging
+import types
+import inspect
+from datetime import datetime
+from pathlib import Path
+from multiprocessing.pool import ThreadPool
+import config
+import pymongo
+from pymongo import MongoClient
+
+
+TIMING_PATH = "timing.csv"
+
+def timefunc(func, counts):
+    ''' Time logging decorator times function execution and logs results
+    :param func: function to execute and time
+    :param counts: reference to counts dictionary to log records affected
+    '''
+    def fncomposite(*args, **kwargs):
+        ''' Timing function called on decorated functions '''
+        start = datetime.now()
+        result = func(*args, **kwargs)
+        end = datetime.now()
+        # Write to log file
+        with open(TIMING_PATH, "a") as file:
+            file.write(f"{func.__name__},{(end - start)},{counts[func.__name__]}\n")
```

Comment:
Nicely done. Some had a difficult time getting the counts in the decorator, this is a good way to do it.